db/migrate/20240807110658_add_index_to_cohorts_name.rb

```diff
@@ -0,0 +1,24 @@
+class AddIndexToCohortsName < ActiveRecord::Migration[7.0]
+  def up
+    # Identify duplicates and update the names.
+    Course.all.find_each do |course|
+      duplicates =
+        course.cohorts.group(:name).having("count(*) > 1").pluck(:name)
+
+      duplicates.each do |name|
+        cohorts = Cohort.where(name: name)
```

Comment:
This should be scoped by the course.
```suggestion
        cohorts = course.cohorts.where(name: name)
```